src/TLuaInterpreter.cpp

```diff
@@ -9482,6 +9482,36 @@ int TLuaInterpreter::getRoomsByPosition(lua_State* L)
     return 1;
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getGridMode
+int TLuaInterpreter::getGridMode(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+    if (!host.mpMap || !host.mpMap->mpRoomDB) {
+        lua_pushnil(L);
+        lua_pushstring(L, "no map present or loaded!");
+        return 2;
+    }
+
+    int id;
+    if (!lua_isnumber(L, 1)) {
+        lua_pushfstring(L, "getGridMode: bad argument #1 type (area id as number expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    } else {
+        id = lua_tonumber(L, 1);
+    }
+
+    TArea* area = host.mpMap->mpRoomDB->getArea(id);
+    if (!area) {
+        lua_pushnil(L);
+        lua_pushfstring(L, "area %s doesn't exist", id);
```

Comment:
`doesn't` ==> `does not` the latter better stresses the *negative* impact of the **not** in the phrase. ("Sorry mate, that area is a figment of your imagination!")

Reply:
I don't think we need the extended negative stress. @Mudlet/ui-review ?

Reply:
I am reluctant for us to use contractions such as *don't*, *shouldn't*, *can't* etc. - it may make it a little harder for the translators to decide whether they have to reproduce such *relaxed* or informal wording in the language they are working in (assuming it has corresponding variations - it may not of course).  I too would be interested in input from the ui-reviewers.

Reply:
You're both members of ui-review team as well :) For me, "doesn't exist" works just fine. It is also a stated fact like "is isn't there", not a suggestion for acting like "don't do X", so no hassle there.

Reply:
You know they're good when they're pedantic :wink: